WordPress/openverse-api
# Add logging and fix all logging contexts

## Fixes


Fixes #776 by @sarayourfriend 

## Description


Add logging to the elasticsearch controller methods.

I also went ahead and updated the logging contexts for all our existing logging in the Django app. To learn more about how this works check out the [`Logget.getChild` docs](https://docs.python.org/3/library/logging.html#logging.Logger.getChild) for more insight into how this works. These changes are necessary to make it easy to discern the context of the logs without having to repeat some context sigil (like the name of the method) in the log messages. This simplifies the code that adds to the log and also ensures a uniform way of filtering logs by context (i.e., method and module name) across the app.

I also added a documentation page describing the approach to logging and the reasoning behind it. Please let me know if any more details should be added here or if the reasons are unclear or otherwise do not make sense.

## Testing Instructions

Checkout this branch. Update your local env to have `LOG_LEVEL` set to `"DEBUG"`. Run the app locally and then run `just logs web`. Test a few endpoints and confirm that the logging works.

Review the added logs and ensure that they are clear and have sufficient context to understand where they're being logged within a method. Remember that the child logger approach ensures that we always know what module and method we're in. We just need to add context (using the code as a map for this context, by the way) enough to be able to tell what's going on from the logs (again, cross referencing with the code). Confirm that I didn't miss any important branches (ideally I've covered all branches except trivial ones).

## Checklist

- [N/A] I added or updated tests for the changes I made (if applicable).

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
